tree/dataframe/src/RDFActionHelpers.cxx

```diff
@@ -233,6 +233,37 @@ template class TakeHelper<float, float, std::vector<float>>;
 template class TakeHelper<double, double, std::vector<double>>;
 #endif
 
+void ValidateSnapshotOutput(const RSnapshotOptions &opts, const std::string &treeName, const std::string &fileName)
+{
+   TString fileMode = opts.fMode;
+   fileMode.ToLower();
+   if (fileMode != "update")
+      return;
```

Comment:
Is the error for file mode `read` handled somewhere else?

Reply:
The _real_ opening of `fileName` (by default in `recreate` mode: it's the output file of the `Snapshot`) is in `SnapshotHelper::Initialize` and `SnapshotHelperMT::Initialize`. We actually do not check that the opening went fine at the moment, but that's outside of the scope of this PR and the bug it tries to fix. That's now https://sft.its.cern.ch/jira/browse/ROOT-10639